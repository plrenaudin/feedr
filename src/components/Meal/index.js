import InputField from "../InputField";
import { inject, observer } from "inferno-mobx";
import FoodCategoryPicker from "../FoodCategoryPicker";

const addItem = (store, meal) => value => {
  store.addItem(meal, value);
};
const removeItem = (store, meal, idx) => () => {
  store.removeItem(meal, idx);
};

const labels = {
  morning: "Breakfast",
  noon: "Lunch",
  snack: "Snack",
  evening: "Diner"
};

const MealItem = ({ index, item, meal, store }) => (
  <li>
    <span>{item.name}</span>
    <div class="actions">
      <FoodCategoryPicker {...{ meal, index }} />
      <a role="button" onClick={removeItem(store, meal, index)}>
        <i class="far fa-trash-alt" />
      </a>
    </div>
  </li>
);

const Meal = ({ meal, store }) => (
  <div class="meal">
    <h3>
      <span class="fa-layers fa-fw">
        <i class="fas fa-circle" data-fa-transform="grow-6" />
        <i class="fas fa-utensils" data-fa-transform="shrink-2" />
      </span>
      {labels[meal]}
    </h3>
    <div class="mealContent">
      <ul $HasKeyedChildren>
        {store.day[meal].map((item, index) => (
          <MealItem key={store.day.date + index} {...{ index, item, meal, store }} />
        ))}
      </ul>
      <InputField
        onChange={addItem(store, meal)}
        placeholder={store.day[meal].length === 0 ? "What did you eat?" : "What else...?"}
      />
    </div>
  </div>
);

export default inject("store")(observer(Meal));
