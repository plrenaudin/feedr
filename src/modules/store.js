import { observable, action, toJS, configure } from "mobx";
import db, { EMPTY_RECORD } from "./db";
import { shortFormatDate } from "./formatter";
import addDays from "date-fns/add_days";

configure({ enforceActions: true });

export default class Store {
  day = observable({ ...EMPTY_RECORD, date: shortFormatDate() });
  constructor() {
    this.load(shortFormatDate());
  }

  set = action(({ date = shortFormatDate(), morning = [], noon = [], evening = [] }) => {
    this.day.date = date;
    this.day.morning = morning;
    this.day.noon = noon;
    this.day.evening = evening;
  });

  addItem = action((meal, data) => {
    this.day[meal].push(data);
    this.save();
  });

  removeItem = action((meal, index) => {
    this.day[meal].splice(index, 1);
    this.save();
  });

  nextDay() {
    this.load(shortFormatDate(addDays(this.day.date, 1)));
  }

  previousDay() {
    this.load(shortFormatDate(addDays(this.day.date, -1)));
  }

  load = action(date => {
    return db.get(date).then(data => {
      this.set(data || { date });
    });
  });

  save() {
    return db.set(this.day.date, toJS(this.day));
  }
}
