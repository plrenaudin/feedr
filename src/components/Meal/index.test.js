import { render } from "inferno";
import Meal from ".";
import store from "../../../test/fixture/store";
import { Provider } from "inferno-mobx";

it("Matches the snapshot", () => {
  const div = document.createElement("div");
  const rendered = render(
    <Provider store={store}>
      <Meal meal="evening" />
    </Provider>,
    div
  );
  expect(rendered.props.children.dom).toMatchSnapshot();
});
