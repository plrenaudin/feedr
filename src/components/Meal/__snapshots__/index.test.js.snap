// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<div
  class="meal"
>
  <h3>
    <span
      class="fa-layers fa-fw"
    >
      <i
        class="fas fa-circle"
        data-fa-transform="grow-6"
      />
      <i
        class="fas fa-utensils"
        data-fa-transform="shrink-2"
      />
    </span>
    Diner
  </h3>
  <div
    class="mealContent"
  >
    <ul>
      <li>
        <span>
          evening Meal 1
        </span>
        <div
          class="actions"
        >
          <div
            class="food-type-picker"
          >
            <a
              role="button"
            >
              <i
                class="fas fa-tag"
              />
            </a>
          </div>
          <a
            role="button"
          >
            <i
              class="far fa-trash-alt"
            />
          </a>
        </div>
      </li>
      <li>
        <span>
          evening Meal 2
        </span>
        <div
          class="actions"
        >
          <div
            class="food-type-picker"
          >
            <a
              role="button"
            >
              <i
                class="fas fa-tag"
              />
            </a>
          </div>
          <a
            role="button"
          >
            <i
              class="far fa-trash-alt"
            />
          </a>
        </div>
      </li>
    </ul>
    <input
      placeholder="What else...?"
      type="text"
    />
  </div>
</div>
`;
